test/unit/Hbar.js

```diff
@@ -2,20 +2,169 @@ import Hbar from "../src/Hbar.js";
 import HbarUnit from "../src/HbarUnit.js";
 
 describe("Hbar", function () {
-    it("should stringify to hbars", function () {
-        expect(new Hbar(10).toString()).to.eql("10 ℏ");
+    it("should pass regex, convert, and reverse [to|from]String", function () {
+        let check = [
+            ["0 ℏ", "0 tℏ"],
+            ["1 tℏ", "1 tℏ"],
+            ["1 μℏ", "100 tℏ"],
+            ["1 mℏ", "0.001 ℏ"],
+            ["1 ℏ", "1 ℏ"],
+            ["1 kℏ", "1000 ℏ"],
+            ["1 Mℏ", "1000000 ℏ"],
+            ["1 Gℏ", "1000000000 ℏ"],
+
+            ["+0 ℏ", "0 tℏ"],
+            ["+1 tℏ", "1 tℏ"],
+            ["+1 μℏ", "100 tℏ"],
+            ["+1 mℏ", "0.001 ℏ"],
+            ["+1 ℏ", "1 ℏ"],
+            ["+1 kℏ", "1000 ℏ"],
+            ["+1 Mℏ", "1000000 ℏ"],
+            ["+1 Gℏ", "1000000000 ℏ"],
+            
+            ["-0 ℏ", "0 tℏ"],
+            ["-1 tℏ", "-1 tℏ"],
+            ["-1 μℏ", "-100 tℏ"],
+            ["-1 mℏ", "-0.001 ℏ"],
+            ["-1 ℏ", "-1 ℏ"],
+            ["-1 kℏ", "-1000 ℏ"],
+            ["-1 Mℏ", "-1000000 ℏ"],
+            ["-1 Gℏ", "-1000000000 ℏ"],
+        ];
+
+        for (let i = 0; i < check.length; i++) {
+            expect(check[i][1]).to.equal(
+                Hbar.fromString(check[i][0]).toString()
+            );
+        }
+    });
+
+    it("should not pass regex", function () {
+        let k = 0;
+        let check = [
+            "1 ",
+            "-1 ",
+            "+1 ",
+            "1.151 ",
+            "-1.151 ",
+            "+1.151 ",
+            "1.",
+            "1.151.",
+            ".1",
+            "1.151 uℏ",
+            "1.151 h",
+        ];
+
+        let i = check.length;
+
+        for (let j = 0; j < check.length; j++) {
+            try {
+                Hbar.fromString(check[j]);
+            } catch (error) {
+                if (error.toString().includes("invalid argument provided")) {
```

Comment:
Perhaps change this to simply be `expect(error.toString().includes("invalid argument provided")).to.be.true;` You'd then be able to remove `i`, `j`, and `k` completely and use a for each loop instead.